src/main/java/io/github/thebusybiscuit/cscorelib2/collections/OptionalBoolean.java

```diff
@@ -0,0 +1,95 @@
+package io.github.thebusybiscuit.cscorelib2.collections;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.NoSuchElementException;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public final class OptionalBoolean {
+
+    private static final OptionalBoolean EMPTY = new OptionalBoolean();
+
+    private final boolean isPresent;
+    private final boolean value;
```

Comment:
Not sure if we could simplify this with a package-private enum here. This would technically allow for 4 possible states whereas there can only be three.

```
EMPTY,
TRUE,
FALSE;
```